### PR TITLE
Methods renamed and return type changed to boolean

### LINKED ISSUE #17701: [Library Stabilization] Should these BigInteger methods return booleans?
There are a number of methods on the `bigint` type that return 0 or some other value.  It seems to me like these methods are a result of booleans being in a weird state for C where they can be used but aren't necessarily used extensively when they could be.  In Chapel, it seems more appropriate to return a bool where possible.

Affected methods:
- `bigint.divisible_p()`
- `bigint.divisible_2exp_p()`
- `bigint.perfect_power_p()`
- `bigint.perfect_square_p()`
- `bigint.congruent_p()`
- `bigint.congruent_2exp_p()`
- `bigint.even_p()`
- `bigint.odd_p()`

I would probably rename these functions to something like `isDivisible` or `isDivisibleBy`, etc. while there, and drop the p (I have no idea why they end in p - maybe it stands for precision?)

Some of these at first glance seem like they would make sense to return something other than a bool, but seem to only return 0 or 1 today, so I think using them in this manner is appropriate (and there's probably something that can be done to ensure they always return the "right" bool value even if the underlying implementation changes (which seems unlikely at this point but is something to watch out for).

### PR DESCRIPTION
**Pull request by tiger-yash**
This PR Partly Resolves #17701 

### Description
* Changed the method names
* Updated the required Deprecation Warnings and tests.

### Methods changed:
 * `bigint.perfect_power_p()`
 * `bigint.perfect_square_p()`
 * `bigint.even_p()`
 * `bigint.odd_p()`

<br>
Note: Rest of the Methods Changed in PR #18784

### CODE REVIEW

File: modules/standard/BigInteger.chpl [hunk 1]
```diff
@@ -2894,7 +2894,18 @@ When ``n/d`` does not produce an integer, this method may produce incorrect resu
...
 
-  proc bigint.perfect_power_p() : int {
+  /*
+    .. warning::
+
+       bigint.perfect_power_p is deprecated, use bigint.isPerfectPower instead
+  */
+  deprecated
+  "bigint.perfect_power_p is deprecated, use bigint.isPerfectPower instead"
+  proc bigint.perfect_power_p() : bool {
```
Comment (lydia-duncan) [inline modules/standard/BigInteger.chpl:2904]: Ditto comment in your other PR about not changing the return type of the deprecated function

File: modules/standard/BigInteger.chpl [hunk 2]
```diff
@@ -2906,10 +2917,24 @@ When ``n/d`` does not produce an integer, this method may produce incorrect resu
       ret = mpz_perfect_power_p(t_.mpz);
     }
 
-    return ret.safeCast(int);
+    if ret.safeCast(int) then
```
Comment (lydia-duncan) [inline modules/standard/BigInteger.chpl:2920]: Ditto comment in your other PR about checking the value of ret rather than if it can safeCast to an int

**Review by lydia-duncan — COMMENTED** [review]
Nice work, just a couple of changes as suggested on the other PR (you'd probably think to do it here on your own, but just in case!)

**Comment by lydia-duncan** [comment]
Note that I don't think these names are as controversial, so I wouldn't worry as much about running them past the mailing list

**Comment by tiger-yash** [comment]
Hey @lydia-duncan!
I think this PR is ready for running tests..

**Review by lydia-duncan — APPROVED** [review]
Nice!  Running the full test suite now

**Comment by lydia-duncan** [comment]
Just two failures (I may start having you run test/standard/GMP/ on your own to catch these :) ):
```
[Error matching program output for library/standard/GMP/ldelaney/gmp_Bigint_misc]
[Error matching program output for library/standard/GMP/ldelaney/gmp_Bigint_powers_roots]
```
Both look like straight-forward updates to which methods are being called

**Pull request merged**